feature/interface/ip/ipv6_slaac_link_local/ipv6_slaac_link_local_test.go

```diff
@@ -0,0 +1,73 @@
+package ipv6_slaac_link_local_test
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"google3/third_party/golang/ygot/ygot/ygot"
+	"google3/third_party/openconfig/featureprofiles/internal/fptest/fptest"
+	"google3/third_party/openconfig/ondatra/gnmi/gnmi"
+	"google3/third_party/openconfig/ondatra/gnmi/oc/oc"
+	"google3/third_party/openconfig/ondatra/ondatra"
+)
+
+var (
+	ipv6BySLAAC     = `fe80::.+/64`
+	intfDesc        = "dutInfSLAAC"
+	waitForAssigned = time.Minute
+	reIPv6BySLAAC   = regexp.MustCompile(ipv6BySLAAC)
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func configureDUTLinkLocalInterface(t *testing.T, dut *ondatra.DUTDevice, p *ondatra.Port) {
+	t.Helper()
+
+	intf := &oc.Interface{Name: ygot.String(p.Name())}
+	intf.Description = ygot.String(intfDesc)
+	intf.GetOrCreateSubinterface(0).GetOrCreateIpv4().SetEnabled(true)
+	intf.GetOrCreateSubinterface(0).GetOrCreateIpv6().SetEnabled(true)
+	gnmi.Replace(t, dut, gnmi.OC().Interface(p.Name()).Config(), intf)
+}
+
+func getAllIPv6Addresses(t *testing.T, dut *ondatra.DUTDevice, p *ondatra.Port) []string {
+	var allIPv6 []string
+	deadline := time.Now().Add(waitForAssigned)
+	for time.Now().Before(deadline) {
+		time.Sleep(10 * time.Second)
+		ipv6Addrs := gnmi.LookupAll(t, dut, gnmi.OC().Interface(p.Name()).Subinterface(0).Ipv6().AddressAny().State())
+		t.Logf("number of ipv6: %d", len(ipv6Addrs))
+		for _, ipv6Addr := range ipv6Addrs {
+			if v6, ok := ipv6Addr.Val(); ok {
+				allIPv6 = append(allIPv6, fmt.Sprintf("%s/%d", v6.GetIp(), v6.GetPrefixLength()))
+			}
+		}
+		if len(allIPv6) > 0 {
```

Comment:
I think this is an ok assumption to make, but mentioning it just for future reference: this func could potentially exit prematurely - in theory - if an implementation reports a non-link-local address before it establishes the link-local address required by RFC4291 (etc).